lib/storage/ldp/ldp-file-store.js

```diff
@@ -0,0 +1,165 @@
+'use strict'
+
+const path = require('path')
+const { LdpFileResource, LdpFileContainer } = require('./ldp-file-resource')
+
+class LdpFileStore {
+  /**
+   * @param options {object}
+   *
+   * @param options.fs {object} Expects the `fs-extra` API
+   * @param options.mapper {LegacyResourceMapper}
+   *
+   * @param options.suffixAcl {string}
+   * @param options.suffixMeta {string}
+   * @param options.dataBrowserPath {string}
+   * @param options.suppressDataBrowser {boolean}
+   */
+  constructor (options) {
+    this.fs = options.fs
+    this.mapper = options.mapper
+
+    this.suffixAcl = options.suffixAcl
+    this.suffixMeta = options.suffixMeta
+    this.dataBrowserPath = options.dataBrowserPath
+    this.suppressDataBrowser = options.suppressDataBrowser
+  }
+
+  /**
+   * @param target {LdpTarget}
+   * @param target.url {string}
+   *
+   * @throws {Error} When encountering a filesystem error that's not "File does
+   *   not exist", such as `EACCES` etc.
+   *
+   * @see https://nodejs.org/api/this.fs.html#fs_class_fs_stats
+   *
+   * @returns {Promise<LdpFileResource|LdpFileContainer>}
+   */
+  async resource (target) {
+    const { path, contentType: mediaType } = await this.mapper.mapUrlToFile(target)
+
+    let exists = true
+    let fsStats
+
+    // Try and load file metadata
+    try {
+      fsStats = await this.fs.stat(path)
```

Comment:
Is this a promise?

Reply:
It is, yeah. See the Promise return type at https://nodejs.org/api/fs.html#fs_filehandle_stat_options